Xamarin.Essentials/Contacts/Contact.ios.cs

```diff
@@ -0,0 +1,143 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Contacts;
+using ContactsUI;
+using Foundation;
+using UIKit;
+
+namespace Xamarin.Essentials
+{
+    public static partial class Contact
+    {
+        internal static Action<PhoneContact> CallBack { get; set; }
+
+        static Task<PhoneContact> PlatformPickContactAsync()
+        {
+            var uiView = Platform.GetCurrentViewController();
+            if (uiView == null)
+                throw new ArgumentNullException($"The View Controller can't be null.");
+
+            using var picker = new CNContactPickerViewController
+            {
+                Delegate = new ContactPickerDelegate()
+            };
+
+            uiView.PresentViewController(picker, true, null);
+            var source = new TaskCompletionSource<PhoneContact>();
+            try
+            {
+                CallBack = (phoneContact) =>
+                {
+                    var tcs = Interlocked.Exchange(ref source, null);
+                    tcs?.SetResult(phoneContact);
+                };
+            }
+            catch (Exception ex)
+            {
+                source.SetException(ex);
+            }
+            return source.Task;
+        }
+
+        internal static PhoneContact GetContact(CNContact contact)
+        {
+            if (contact == null)
+                return default;
+
+            try
+            {
+                var contactType = ToPhoneContact(contact.ContactType);
+                var phones = new Dictionary<string, ContactType>();
+
+                foreach (var item in contact.PhoneNumbers)
+                    phones.Add(item?.Value?.StringValue, contactType);
+
+                var emails = new Dictionary<string, ContactType>();
+
+                foreach (var item in contact.EmailAddresses)
+                    emails.Add(item?.Value?.ToString(), contactType);
+
+                var name = $"{contact.GivenName} {contact.MiddleName} {contact.FamilyName}";
```

Comment:
Should check for empty string in `MiddleName`, to prevent generating a full name consisting of double spaces between `GivenName` and `FamilyName`.

If the contact type is `CNContactType.Organization`, the `OrganizationName` property should be used as the contact name to prevent returning a contact without a name.